CryptoTechReminderSystem.Test/SlackGatewayTests.cs

```diff
@@ -0,0 +1,99 @@
+using System.Linq;
+using CryptoTechReminderSystem.DomainObject;
+using CryptoTechReminderSystem.Gateway;
+using FluentAssertions;
+using FluentSim;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using static Newtonsoft.Json.JsonConvert;
+
+namespace CryptoTechReminderSystem.Test
+{
+    public class SlackGatewayTests
+    {
+        private FluentSimulator _fluentSimulator;
+        
+        public class SlackPostMessageResponse
+        {
+            [JsonProperty("ok")] public bool IsOk;
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            _fluentSimulator = new FluentSimulator("http://localhost:8011/");
+            
+            var slackPostMessageResponse = new SlackPostMessageResponse
+            {
+                IsOk = true
+            };
+            
+            _fluentSimulator.Post("/api/chat.postMessage").Responds(slackPostMessageResponse);
+
+            _fluentSimulator.Start();
+        }
+        
+        [TearDown]
+        public void TearDown()
+        {
+            _fluentSimulator.Stop();
+        }
+        
+        [Test]
+        public void CanSendAPostMessageRequest()
+        {
+            var messageSender = new SlackGateway("http://localhost:8011/", "xxxx-xxxxxxxxx-xxxx");
+            
+            messageSender.Send(new Message());
+            
+            var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+                
+            receivedRequest.Url.Should().Be("http://localhost:8011/api/chat.postMessage");
+        }
+        
+        [Test]
+        public void CanSendAPostMessageRequestWithAToken()
+        {
+            var messageSender = new SlackGateway("http://localhost:8011/", "xxxx-xxxxxxxxx-xxxx");
+            
+            messageSender.Send(new Message());
+            
+            var receivedRequest = _fluentSimulator.ReceivedRequests.First();
+                
+            receivedRequest.Url.Should().Be("http://localhost:8011/api/chat.postMessage");
```

Comment:
Can this test test that the request has a token?

Reply:
This has been fixed in 8c9d555b573e517efd5d9ad5be19322482078acd.